rocon_python_comms/src/rocon_python_comms/connections.py

```diff
@@ -137,42 +156,56 @@ def msg(self):
     def msg(self, msg):
         self._connection = msg
 
-    def generate_type_info(self):
-        '''
+    def generate_type_info_msg(self):
+        """
         Basic connection details are provided by get system state from the master, which is
         a one shot call to give you information about every connection possible. it does
         not however provide type info information and the only way of retrieving that from
         the master is making one xmlrpc call to the master for every single connection.
         This gets expensive, so generating this information is usually delayed until we
         need it and done via this method.
-        '''
+        """
         if self.type_info is None:
             if self.type == PUBLISHER or self.type == SUBSCRIBER:
-                self.type_info = rostopic.get_topic_type(self.name)[0]  # message type
+                try:
+                    self.type_info = rostopic.get_topic_type(self.name)[0]  # message type
+                    self.type_msg = self.type_info
+                except rostopic.ROSTopicIOException as topic_exc:
+                    rospy.logwarn(topic_exc)
             elif self.type == SERVICE:
-                self.type_info = rosservice.get_service_uri(self.name)
+                try:
+                    self.type_info = rosservice.get_service_uri(self.name)
+                    self.type_msg = rosservice.get_service_type(self.name)
+                except rosservice.ROSServiceIOException as service_exc:
+                    rospy.logwarn(service_exc)
```

Comment:
Can you prefix this guy, something like:

```
rospy.logwarn("Connection: %s" % service_exc)
```

as it gets very hard to track down which component is throwing obscure messages.
```